core/state/statedb.go

```diff
@@ -514,12 +529,20 @@ func (s *StateDB) updateStateObject(obj *stateObject) {
 		s.setError(fmt.Errorf("updateStateObject (%x) error: %v", addr[:], err))
 	}
 
-	// If state snapshotting is active, cache the data til commit. Note, this
-	// update mechanism is not symmetric to the deletion, because whereas it is
-	// enough to track account updates at commit time, deletions need tracking
-	// at transaction boundary level to ensure we capture state clearing.
-	if s.snap != nil {
```

Comment:
From myview, seem they expect that snap layer always enables when initing Statedb, so accounts here still points to snapAccounts as a cache layer, just renaming variable.